client/jetpack-connect/controller.js

```diff
@@ -96,54 +87,40 @@ export default {
 		next();
 	},
 
-	personal( context ) {
-		const analyticsBasePath = '/jetpack/connect/personal',
-			analyticsPageTitle = 'Jetpack Connect Personal';
-
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
-
-		jetpackConnectFirstStep( context, 'personal' );
-	},
-
-	premium( context ) {
-		const analyticsBasePath = '/jetpack/connect/premium',
-			analyticsPageTitle = 'Jetpack Connect Premium';
-
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
-
-		jetpackConnectFirstStep( context, 'premium' );
-	},
-
 	newSite( context ) {
 		analytics.pageView.record( '/jetpack/new', 'Add a new site (Jetpack)' );
 		jetpackNewSiteSelector( context );
 	},
 
-	pro( context ) {
-		const analyticsBasePath = '/jetpack/connect/pro',
-			analyticsPageTitle = 'Jetpack Install Pro';
-
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
-
-		jetpackConnectFirstStep( context, 'pro' );
-	},
-
-	install( context ) {
-		const analyticsBasePath = '/jetpack/connect/install',
-			analyticsPageTitle = 'Jetpack Install';
+	connect( context ) {
+		const {
+			path,
+			pathname,
+			params
+		} = context;
+		const { type = false } = params;
+		const analyticsPageTitle = get( type, analyticsPageTitleByType, 'Jetpack Connect' );
 
-		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
+		debug( 'entered connect flow with params %o', params );
 
-		jetpackConnectFirstStep( context, 'install' );
-	},
+		analytics.pageView.record( pathname, analyticsPageTitle );
 
-	connect( context ) {
-		const analyticsBasePath = '/jetpack/connect',
-			analyticsPageTitle = 'Jetpack Connect';
+		removeSidebar( context );
```

Comment:
This call (indeed the function) may not be required at all, since we specify `secondary: false` for the [whole section](https://github.com/Automattic/wp-calypso/blob/31e2ca4f043f2047803735e94c7355e39f8d6af4/client/wordpress-com.js#L142).